lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -179,6 +179,34 @@ def self.fetch_all_tags
         Action.sh('git', 'fetch', '--tags')
       end
 
+      # Calculates if there are new commits between two references in the Git repository.
+      #
+      # @param base_ref [String] The base reference (branch name, tag, or commit hash).
+      # @param head_ref [String] The head reference (branch name, tag, or commit hash).
+      # @param first_parent [Boolean] Counts only the first parent commit upon seeing a merge commit
+      #
+      # @return [Boolean] Whether there are new commits between base_ref and head_ref.
+      #
+      def self.has_commits_between?(base_ref:, head_ref:, first_parent: true)
+        git_repo = Git.open(Dir.pwd)
+
+        return git_repo.log.between(base_ref, head_ref).count.positive? unless first_parent
+
+        base_commit = git_repo.gcommit(base_ref)
+        head_commit = git_repo.gcommit(head_ref)
+
+        count = 0
+        current_commit = head_commit
+        while current_commit
+          break if current_commit.sha == base_commit.sha
+
+          count += 1
+          current_commit = current_commit.parents.first
+        end
+
+        count.positive?
+      end
```

Comment:
I'm not sure we'll ever need the differentiation of `first_parent:` parameter anymore now that the goal is to just testing if the value is `.positive?` and return a boolean. So I think we can remove the parameter, and all the code after `unless first_parent` too now. Because instead of counting the total number then testing if it is `positive?`, it would be way more effective to exit early as soon as we detect they are not the same commits, instead of walking up the `current_commit = current_commit.parents.first` chain only to test if `count.positive?` in the end.

Also, I just realized that that walk up the chain (in the case of `first_parent: true`) could risk entering an infinite (well, or super-long-until-we-reach-the-repo-root) loop if we happen to call `has_commits_between?` with refs which are unreachable from one another. For example if we were to call `has_commits_between?(D, F)` in [your example of such a git tree](https://github.com/wordpress-mobile/release-toolkit/pull/570/files#diff-7aed954270eabe63c0117e184d02a920e7c50e090136cd8a33fe121bd7e87ca9R118-R122)

So in conclusion I think this whole code can be simplified to:

```suggestion
      # @return [Boolean] Whether two git refs point to the same commit
      #
      def self.point_to_same_commit?(ref1, ref2)
        git_repo = Git.open(Dir.pwd)

        commit1 = git_repo.gcommit(ref1)
        commit2 = git_repo.gcommit(ref2)

        return commit1.sha == commit2.sha
      end
```

And then in your action implementation, you'd call `if point_to_same_commit?(base_branch, head_branch)` instead of `unless has_commits_between?(base_ref: base_branch, head_ref: head_branch)`

Reply:
Yeah, I think for the purpose of it, probably this line would've been enough:
```ruby
git_repo.log.between(base_ref, head_ref).count.positive?
```

Though I've updated `point_to_same_commit?`, what it does is a bit more straight forward. Updated on 84f3edd7b8221ab8cab2214f1634ee259aae1ff4